### PR TITLE
[dart-dio] Adds support for multipart form data post body

### PR DESCRIPTION
**Pull request by athornz**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC: @ircecho @swipesight @jaumard

### CODE REVIEW

**Comment by wing328** [comment]
cc @nickmeinhold (2019/09)  @amondnet (2019/12) as well

File: modules/openapi-generator/src/main/resources/dart-dio/api.mustache [hunk 1]
```diff
@@ -25,23 +25,38 @@ class {{classname}} {
...
-                "{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+        List<String> contentTypes = [{{#consumes}}"{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+
+
+        {{#formParams}}
+        Map<String, dynamic> formData = {};
+        {{#isMultipart}}
+            {{^isFile}}
+                if ({{paramName}} != null) {
+                    formData['{{baseName}}'] = parameterToString({{paramName}});
```
Comment (amondnet) [inline modules/openapi-generator/src/main/resources/dart-dio/api.mustache]: @athornz `parameterToString` isn't defined.

File: modules/openapi-generator/src/main/resources/dart-dio/api.mustache [hunk 2]
```diff
@@ -25,23 +25,38 @@ class {{classname}} {
...
+        queryParams.removeWhere((key, value) => value == null);
+        headerParams.removeWhere((key, value) => value == null);
 
-            List<String> contentTypes = [{{#consumes}}
-                "{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+        List<String> contentTypes = [{{#consumes}}"{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
+
+
+        {{#formParams}}
+        Map<String, dynamic> formData = {};
```
Comment (amondnet) [inline modules/openapi-generator/src/main/resources/dart-dio/api.mustache]: @athornz  If there are multiple form parameters, `Map<String, dynamic> formData = {};` will be generated in duplicate.

Comment (athornz) [reply modules/openapi-generator/src/main/resources/dart-dio/api.mustache]: ah yes good point

**Comment by wing328** [comment]
I believe @athornz has already addressed the feedback via https://github.com/OpenAPITools/openapi-generator/pull/4797/commits/a0ac34b374b34d6224adf5535a0ee32b402140d1 so I'll go ahead to merge this PR into master.

**Pull request merged**